### PR TITLE
New feature : Csv Export 

### PR DESCRIPTION
**Pull request by Ajeett01**
# Description

Relates to issue: **ZEUS-#1467**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "fetch-libraries": "bash fetch-libraries.sh"
   },
   "dependencies": {
+    "@json2csv/plainjs": "^7.0.6",
```
Comment (shubhamkmr04) [inline package.json:34]: Lock the version here (remove `^` and run yarn `again`)

File: package.json [hunk 2]
```diff
@@ -156,6 +157,7 @@
     "@babel/plugin-proposal-nullish-coalescing-operator": "7.18.6",
     "@babel/plugin-proposal-optional-chaining": "7.21.0",
     "@babel/plugin-transform-arrow-functions": "7.22.5",
+    "@babel/plugin-transform-export-namespace-from": "^7.24.7",
```
Comment (shubhamkmr04) [inline package.json:160]: here too

**Comment by shubhamkmr04** [comment]
We might have to change the file name from `JsonToCsv.tsx` to something else. 
@kaloudis your thoughts?

File: views/Activity/JsonToCsv.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        >
+            <View style={styles.modalOverlay}>
+                <View style={styles.modalView}>
+                    <TextInput
+                        placeholder="File name (optional)"
+                        value={customFileName}
+                        onChangeText={setCustomFileName}
+                    />
+                    <View>
+                        <Button title="Download CSV" onPress={downloadCsv} />
```
Comment (kaloudis) [inline views/Activity/JsonToCsv.tsx:133]: this needs to be locatlized

**Comment by kaloudis** [comment]
> We might have to change the file name from `JsonToCsv.tsx` to something else. @kaloudis your thoughts?

`ActivityToCsv`, perhaps?

File: views/Activity/JsonToCsv.tsx (hunk 3 above)
Comment (shubhamkmr04) [reply views/Activity/JsonToCsv.tsx:133]: @Ajeett01 look through [locale files](https://github.com/ZeusLN/zeus/blob/master/locales/en.json) and add your used strings here

**Comment by shubhamkmr04** [comment]
> > We might have to change the file name from `JsonToCsv.tsx` to something else. @kaloudis your thoughts?
> 
> `ActivityToCsv`, perhaps?

Yup we can, or maybe `ImportActivity`, lets iterate on that

File: views/Activity/Activity.tsx [hunk 4]
```diff
@@ -225,6 +243,21 @@ export default class Activity extends React.PureComponent<
...
+            <TouchableOpacity
+                onPress={this.handleDownloadPress}
+                accessibilityLabel={localeString('general.download')}
+            >
+                <Icon
+                    name="download"
+                    type="feather"
+                    color={themeColor('text')}
+                    underlayColor="transparent"
+                    size={30}
```
Comment (shubhamkmr04) [inline views/Activity/Activity.tsx:256]: adjust the download icon size here as compared to filter and cross icons

File: babel.config.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
...
-    'react-native-reanimated/plugin'
-  ]
-}
\ No newline at end of file
+    presets: ['module:@react-native/babel-preset'],
+    plugins: [
+        ['@babel/plugin-proposal-decorators', { legacy: true }],
+        'react-native-reanimated/plugin',
+        ['@babel/plugin-transform-export-namespace-from']
+    ]
```
Comment (shubhamkmr04) [inline babel.config.js:7]: why have we added that?

File: views/Activity/JsonToCsv.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            return '';
+        }
+
+        const keysToInclude = [
+            'amt_paid',
+            'amt_paid_sat',
+            'cltv_expiry',
+            'creation_date',
+            'expiry'
+        ];
```
Comment (shubhamkmr04) [inline views/Activity/JsonToCsv.tsx:66]: @kaloudis would these keys be good to download in our csv data?

Comment (kaloudis) [reply views/Activity/JsonToCsv.tsx:66]: these look ok, definitely need to flesh them out though

File: views/Activity/ActivityToCsv.tsx [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    const requestAndroidPermissions = async () => {
+        try {
+            const granted = await PermissionsAndroid.request(
+                PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+                {
+                    title: 'Storage Permission Required',
+                    message: 'This app needs access to your storage to save CSV files',
+                    buttonNeutral: 'Ask Me Later',
+                    buttonNegative: 'Cancel',
+                    buttonPositive: 'OK'
```
Comment (kaloudis) [inline views/Activity/ActivityToCsv.tsx:109]: We should localize these strings

**Comment by shubhamkmr04** [comment]
@Ajeett01 We need to fix failing check on the latest commit

File: views/Activity/ActivityToCsv.tsx [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+                    .join(',');
+                const rows = transactionData
+                    .map((item) =>
+                        transactionKeysToInclude
+                            .map((field) => item[field.value] || '')
+                            .join(',')
+                    )
+                    .join('\n');
+
+                return `Transaction Section\n${header}\n${rows}`;
```
Comment (kaloudis) [inline views/Activity/ActivityToCsv.tsx:153]: this should be localized

File: views/Activity/ActivityToCsv.tsx [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+
+                const invoiceHeader = [
+                    'Amount Paid',
+                    'Amount Paid (Sat)',
+                    'CLTV Expiry',
+                    'Creation Date',
+                    '',
+                    '',
+                    'Expiry'
+                ].join(',');
```
Comment (kaloudis) [inline views/Activity/ActivityToCsv.tsx:170]: these should be localized

File: views/Activity/ActivityToCsv.tsx [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+
+                const invoiceSection = [
+                    'Invoice Section',
```
Comment (kaloudis) [inline views/Activity/ActivityToCsv.tsx:173]: this needs localization

File: views/Activity/ActivityToCsv.tsx [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+                    invoiceHeader,
+                    ...invoiceData.map((item) =>
+                        invoiceKeysToInclude
+                            .map((field) => item[field.value] || '')
+                            .join(',')
+                    )
+                ].join('\n');
+
+                const paymentSection = [
+                    'Payment Section',
```
Comment (kaloudis) [inline views/Activity/ActivityToCsv.tsx:183]: this needs localization

File: views/Activity/Activity.tsx [hunk 12]
```diff
@@ -132,6 +135,21 @@ export default class Activity extends React.PureComponent<
         return 'secondaryText';
     };
 
+    handleDownloadPress = () => {
+        const { ActivityStore } = this.props;
+        const { startDate, endDate } = ActivityStore.filters;
+        const { filteredActivity } = ActivityStore;
+
+        if (!startDate || !endDate) {
+            Alert.alert('Error', 'Please select Start and End date.');
```
Comment (kaloudis) [inline views/Activity/Activity.tsx:144]: this needs localization

File: babel.config.js (hunk 5 above)
Comment (kaloudis) [reply babel.config.js:7]: I am curious about this too

**Review by kaloudis — CHANGES_REQUESTED** [review]

**Comment by kaloudis** [comment]
closing in favor of https://github.com/ZeusLN/zeus/pull/2436

**Pull request closed without merging**